### PR TITLE
URIs validation to protect from running external resources

### PR DESCRIPTION
**Pull request by Alroniks**
Task is https://oat-sa.atlassian.net/browse/TAO-9867

### CODE REVIEW

File: model/offline/JsonOfflineTestParser.php [hunk 1]
```diff
@@ -76,8 +89,60 @@ protected function getContent()
...
+     *
+     * @return bool
+     */
+    public function checkUri($uri)
+    {
+        /** @var UriProvider $uriProvider */
+        $uriProvider = $this->getUriProviderService();
+        $validInstanceLink = $uriProvider->provide();
+
+//        print_r($uri);
```
Comment (SergiiTao) [inline model/offline/JsonOfflineTestParser.php]: please remove comments

Comment (Alroniks) [reply model/offline/JsonOfflineTestParser.php]: Oops 😮

File: model/offline/JsonOfflineTestParser.php [hunk 2]
```diff
@@ -76,8 +89,60 @@ protected function getContent()
...
+    {
+        $body = $this->getBody();
+
+        $validateUriFunction = [$this, 'checkUri'];
 
+        $invalid = 0;
+
+        array_walk_recursive($body, static function ($item, $key) use ($validateUriFunction, &$invalid) {
+            if (in_array($key, self::PROTECTED_FIELDS, true) && !$validateUriFunction($item)) {
+                $invalid++;
```
Comment (SergiiTao) [inline model/offline/JsonOfflineTestParser.php]: I'd raise an exception just right here

File: model/offline/JsonOfflineTestParser.php [hunk 3]
```diff
@@ -76,8 +89,60 @@ protected function getContent()
...
+    public function validateUris()
+    {
+        $body = $this->getBody();
+
+        $validateUriFunction = [$this, 'checkUri'];
 
+        $invalid = 0;
+
+        array_walk_recursive($body, static function ($item, $key) use ($validateUriFunction, &$invalid) {
+            if (in_array($key, self::PROTECTED_FIELDS, true) && !$validateUriFunction($item)) {
```
Comment (SergiiTao) [inline model/offline/JsonOfflineTestParser.php]: this algorithm doesn't take to account nesting level of a field. is it OK?

Comment (Alroniks) [reply model/offline/JsonOfflineTestParser.php]: In docs said "Applies the user-defined callback function to each element of the array. This function will recurse into deeper arrays." that means that this function will go to nested levels.
https://www.php.net/manual/en/function.array-walk-recursive.php

File: model/offline/JsonOfflineTestParser.php (hunk 2 above)
Comment (Alroniks) [reply model/offline/JsonOfflineTestParser.php]: Hm, probably it makes sense

**Pull request merged**